### PR TITLE
Optimize Component::hasEventHandlers(), Component::trigger()

### PR DESCRIPTION
**Pull request by WinterSilence**
Quick/easy checks are performed first

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

**Review by bizley — APPROVED** [review]

File: framework/base/Component.php [hunk 1]
```diff
@@ -594,10 +598,12 @@ public function off($name, $handler = null)
...
      * Triggers an event.
-     * This method represents the happening of an event. It invokes
-     * all attached handlers for the event including class-level handlers.
+     *
+     * This method represents the happening of an event. It invokes all attached handlers for the event
+     * including class-level handlers.
+     *
      * @param string $name the event name
-     * @param Event $event the event parameter. If not set, a default [[Event]] object will be created.
+     * @param Event|null $event the event instance. If ommit, a default [[Event]] object will be created.
```
Comment (samdark) [inline framework/base/Component.php:606]:
```suggestion
     * @param Event|null $event the event instance. If not set, a default [[Event]] object will be created.
```

**Review by samdark — APPROVED** [review]
Please add a line for CHANGELOG.

**Comment by WinterSilence** [comment]
@samdark done

**Comment by samdark** [comment]
:+1

**Pull request merged**